### DISCUSSION TITLE
Why sendAll deprecated

### DISCUSSION THREAD

**Discussion by thomasglachant** [firebase/firebase-admin-node#2470] [Q&A]
Hi! 

I work on an application that need to send a notifications to a lot of tokens (more than 1000). 

I saw that the `sendAll` method is deprecated and replaced by the `sendEach` method. 
This method send the messages one by one, with poorly performance than the `sendAll` method

Do anyone know why the sendAll method should not be used anymore ?

**Comment by pquerner** [comment]
See this answer: https://github.com/firebase/firebase-admin-node/issues/2418#issuecomment-1989060437
